csrc/python_frontend/python_bindings.cpp

```diff
@@ -3149,6 +3157,31 @@ void initNvFuserPythonBindings(PyObject* module) {
       py::arg("pos") = -1,
       py::arg("best_effort") = false,
       py::arg("selected_tensors") = std::vector<Tensor>());
+  nvf_sched.def("tensors", [](FusionDefinition::SchedOperators& self) {
+    NVF_CHECK(
+        self.validUse(),
+        "Attempting to use a SchedOperators Op prior to definition!");
+    // Return all Tensors in FusionDefinition
+    return self.fusion_definition->tensors();
+  });
+  nvf_sched.def(
+      "is_reduction",
+      [](FusionDefinition::SchedOperators& self, Tensor tensor) {
+        NVF_CHECK(
+            self.validUse(),
+            "Attempting to use a SchedOperators Op prior to definition!");
+        // Determine if tensor is a result from a reduction operation.
+        FusionDefinition* fd = self.fusion_definition;
+        TensorView* tv =
+            fd->getFusionState(tensor.index)->template as<TensorView>();
+        return (
+            !tv->isFusionInput() &&
```

Comment:
out of curiosity, here we do not have segmentation, so technically, we should never see inputs with reduction id?

Reply:
I'm interested in adding segmentation. I'll leave it here to avoid missing it in the future.